ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+# lower uint types - which should be avoided!)
+# {i: min((dtype for dtype, maxsize in REPRESENTABLE_FLAGS.items()
+#          if maxsize>=i),
+#         key=REPRESENTABLE_FLAGS.get)
+#  for i in range(65)}
+_LAYERS_TO_DTYPE = {
+    0:  'int8',   1: 'int8',   2: 'int8',   3: 'int8',   4: 'int8',
+    5:  'int8',   6: 'int8',   7: 'int8',   8: 'int16',  9: 'int16',
+    10: 'int16', 11: 'int16', 12: 'int16', 13: 'int16', 14: 'int16',
+    15: 'int16', 16: 'int32', 17: 'int32', 18: 'int32', 19: 'int32',
+    20: 'int32', 21: 'int32', 22: 'int32', 23: 'int32', 24: 'int32',
+    25: 'int32', 26: 'int32', 27: 'int32', 28: 'int32', 29: 'int32',
+    30: 'int32', 31: 'int32', 32: 'int64', 33: 'int64', 34: 'int64',
+    35: 'int64', 36: 'int64', 37: 'int64', 38: 'int64', 39: 'int64',
+    40: 'int64', 41: 'int64', 42: 'int64', 43: 'int64', 44: 'int64',
+    45: 'int64', 46: 'int64', 47: 'int64', 48: 'int64', 49: 'int64',
+    50: 'int64', 51: 'int64', 52: 'int64', 53: 'int64', 54: 'int64',
+    55: 'int64', 56: 'int64', 57: 'int64', 58: 'int64', 59: 'int64',
+    60: 'int64', 61: 'int64', 62: 'int64', 63: 'int64', 64: 'uint64'
+}
+
+
+class Bitmask(object):
+    """Bitmask representation that can be converted to normal masks.
+
+    Parameters
+    ----------
+    bitmask : `numpy.ndarray` of integer data type
+        The bitmask.
+
+    name_map : sequence or mapping or None, optional
+        Provide names for the different flags of the bitmask. This allows
+        accessing the flags by name rather than by position or value of the
+        bit. If it's a sequence then the names should be listed by increasing
+        bitvalue. If it's a mapping the names should be the keys and the
+        bit position should be the key.
+
+        For example if ``1`` represents ``bad pixel`` and ``2``
+        represents ``hot pixel`` then the ``name_map`` should be either:
+        ``["bad pixel", "hot pixel"]`` or ``{"bad pixel": 0, "hot pixel": 1}``
+
+        Default is ``None``.
+
+    Examples
+    --------
+    Creating an unnamed bitmask::
+
+        >>> import numpy as np
+        >>> from ccdproc import Bitmask
+        >>> Bitmask(np.arange(10).reshape(2,5))
+        Bitmask (
+        array([[0, 1, 2, 3, 4],
+               [5, 6, 7, 8, 9]])
+        name_map={}
+        )
+
+    Or creating a named bitmask::
+
+        >>> Bitmask(np.arange(10),
+        ...         name_map=['bad', 'hot', 'artifact', 'low response'])
+        Bitmask (
+        array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        name_map=OrderedDict([('bad', 0), ('hot', 1), ('artifact', 2), ('low response', 3)])
+        )
+
+        >>> names = {'bad': 0, 'hot': 1, 'artifact': 2, 'low response': 3}
+        >>> bm = Bitmask(np.arange(10), name_map=names)
+        >>> bm  # doctest: +SKIP
+        Bitmask (
+        array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        name_map={'hot': 1, 'low response': 3, 'bad': 0, 'artifact': 2}
```

Comment:
Maybe @mcara can explain on the specifics of AstroDrizzle usage. Would be nice if the API is somewhat consistent between the two, as HST users are already probably used to AstroDrizzle API (or maybe not if they only use it like a black box).